sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -164,4 +164,28 @@ case class HiveTableScanExec(
   }
 
   override def output: Seq[Attribute] = attributes
+
+  override def sameResult(plan: SparkPlan): Boolean = plan match {
+    case other: HiveTableScanExec =>
+      val thisRequestedAttributes = requestedAttributes.map(cleanExpression)
+      val otherRequestedAttributes = other.requestedAttributes.map(cleanExpression)
+
+      val result = partitionPruningPred == other.partitionPruningPred &&
+        relation.sameResult(other.relation) &&
+          thisRequestedAttributes.zip(otherRequestedAttributes)
+            .forall(p => p._1.semanticEquals(p._2))
+      result
+    case _ => false
+  }
+
+  private def cleanExpression(e: Attribute): Expression = e match {
+    case a: AttributeReference =>
+      // As the root of the expression, Alias will always take an arbitrary exprId, we need
```

Comment:
this comment doesn't match the code. Can you explain more about why the default `cleanExpression` doesn't work?